ssl/t1_lib.c

```diff
@@ -1164,29 +1319,45 @@ int ssl_setup_sig_algs(SSL_CTX *ctx)
          * could be that the signature is available, and the hash is available
          * independently - but not as a combination. We ignore this for now.
          */
-        if (lu->hash != NID_undef
-                && ctx->ssl_digest_methods[lu->hash_idx] == NULL) {
+        if (cache[i].hash != NID_undef
+                && ctx->ssl_digest_methods[cache[i].hash_idx] == NULL) {
             cache[i].enabled = 0;
             continue;
         }
 
-        if (!EVP_PKEY_set_type(tmpkey, lu->sig)) {
+        if (cache[i].name) {
+            pctx = EVP_PKEY_CTX_new_from_name(ctx->libctx, cache[i].name, ctx->propq);
+            if (pctx) {
+                EVP_PKEY_CTX_free(pctx);
+                continue;
+            }
+        }
+
+        if (!EVP_PKEY_set_type(tmpkey, cache[i].sig)) {
             cache[i].enabled = 0;
             continue;
         }
+
         pctx = EVP_PKEY_CTX_new_from_pkey(ctx->libctx, tmpkey, ctx->propq);
-        /* If unable to create pctx we assume the sig algorithm is unavailable */
-        if (pctx == NULL)
+        /* If unable to create pctx from either name or pkey, we assume the sig algorithm is unavailable. */
+        if (!pctx)
```

Comment:
Keep this as `pctx == NULL`

Reply:
ah...oops...sure.  I thought I had written that bit and was trying to conform to the surrounding code.